crates/accelerate/src/sparse_observable.rs

```diff
@@ -348,420 +219,16 @@ pub enum LabelError {
-///
-///
-/// Mathematical manipulation
-/// =========================
-///
-/// :class:`SparseObservable` supports the standard set of Python mathematical operators like other
-/// :mod:`~qiskit.quantum_info` operators.
-///
-/// In basic arithmetic, you can:
-///
-/// * add two observables using ``+``
-/// * subtract two observables using ``-``
-/// * multiply or divide by an :class:`int`, :class:`float` or :class:`complex` using ``*`` and ``/``
-/// * negate all the coefficients in an observable with unary ``-``
-///
-/// Each of the basic binary arithmetic operators has a corresponding specialized in-place method,
-/// which mutates the left-hand side in-place.  Using these is typically more efficient than the
-/// infix operators, especially for building an observable in a loop.
-///
-/// The tensor product is calculated with :meth:`tensor` (for standard, juxtaposition ordering of
-/// Pauli labels) or :meth:`expand` (for the reverse order).  The ``^`` operator is overloaded to be
-/// equivalent to :meth:`tensor`.
-///
-/// .. note::
-///
-///     When using the binary operators ``^`` (:meth:`tensor`) and ``&`` (:meth:`compose`), beware
-///     that `Python's operator-precedence rules
-///     <https://docs.python.org/3/reference/expressions.html#operator-precedence>`__ may cause the
-///     evaluation order to be different to your expectation.  In particular, the operator ``+``
-///     binds more tightly than ``^`` or ``&``, just like ``*`` binds more tightly than ``+``.
-///
-///     When using the operators in mixed expressions, it is safest to use parentheses to group the
-///     operands of tensor products.
-///
-/// A :class:`SparseObservable` has a well-defined :meth:`adjoint`.  The notions of scalar complex
-/// conjugation (:meth:`conjugate`) and real-value transposition (:meth:`transpose`) are defined
-/// analogously to the matrix representation of other Pauli operators in Qiskit.
-///
-///
-/// Efficiency notes
-/// ----------------
-///
-/// Internally, :class:`SparseObservable` is in-place mutable, including using over-allocating
-/// growable vectors for extending the number of terms.  This means that the cost of appending to an
-/// observable using ``+=`` is amortised linear in the total number of terms added, rather than
-/// the quadratic complexity that the binary ``+`` would require.
-///
-/// Additions and subtractions are implemented by a term-stacking operation; there is no automatic
-/// "simplification" (summing of like terms), because the majority of additions to build up an
-/// observable generate only a small number of duplications, and like-term detection has additional
-/// costs.  If this does not fit your use cases, you can either periodically call :meth:`simplify`,
-/// or discuss further APIs with us for better building of observables.
-#[pyclass(module = "qiskit.quantum_info", sequence)]
+/// See PySparseObservable in crates/accelerate/src/py_sparse_observable for detailed docs.
```

Comment:
Possibley

> See `[PySparseObservable]` for detailed docs.

might be clearer that it's just implied you should go find that struct, and less likely for the file to get out of sync.